Reorganized the Close Menu, Quit, and Reset buttons to match the layout in SoH.

Glought opened the pull request:
I reorganized the menu buttons to match the placement and order used in SoH. 

I also port over the SohModals system and renamed it to BenModals. With this change, when the player presses the Quit button, a confirmation message will appear asking, ‘Are you sure you want to quit 2S2H?’ with two options: Quit and Cancel, like how it works in SoH.

Before:
<img width="1826" height="116" alt="2ship Original Menu Buttons" src="https://github.com/user-attachments/assets/8e662a74-f18d-4d5b-adfd-013399dc9730" />

After:
<img width="1914" height="211" alt="2ship reorganized Menu Buttons" src="https://github.com/user-attachments/assets/a8f19c95-372d-4060-be30-a3e189e15475" />



<!--- section:artifacts:start -->
### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/4584058765.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/4584063122.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/4584118528.zip)
<!--- section:artifacts:end -->

Glought committed:
Reorganized the Menu "Reset","Quit 2S2H , and "Closed Menu" buttons
Place in the same order as it is in SoH "Quit, Reset, and Close Menu" on
the right.

Glought committed:
Ported "SohModals" from SoH and Renamed it to "BenModals

Glought committed:
When the "Quit" Button is pressed it will pop up a message asking the
Player "Are you sure you want to quit 2S2H?". with two buttons
"Quit" and "Cancel" like how it is done in SoH.

Eblo commented on mm/2s2h/BenGui/BenModals.h:
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <libultraship/libultraship.h>
```
Do you need this entire header? This is a rather hefty one that is not commonly used directly.

Eblo reviewed (commented):
Nice. No complaints, just one blocking question. Modals are a welcome addition as well.

Eblo commented on mm/2s2h/BenGui/BenModals.cpp:
```diff
@@ -0,0 +1,87 @@
+#include <vector>
+#include <string>
+#include <libultraship/bridge.h>
+#include <libultraship/libultraship.h>
```
Ditto, but also including the same header here is redundant when `BenModals.h` includes it as well.

Glought committed:
Cleaned up unneeded "includes" in "BenModals.h" and "BenModals.cpp"
Removed "extern "C" PlayState* gPlayState;" from "BenModals.cpp"
it wasn't used for anything.

Glought commented:
I cleaned up the unneeded `"Includes` from `BenModals.h` and `BenModals.cpp` .

Eblo committed:
Merge branch 'develop' into reorganize_Close_Quit_Reset_Menu_Buttons
